internal/extproc/translator/openai_gcpanthropic_stream.go

```diff
@@ -0,0 +1,357 @@
+// Copyright Envoy AI Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package translator
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/anthropics/anthropic-sdk-go"
+	"github.com/anthropics/anthropic-sdk-go/shared/constant"
+	extprocv3 "github.com/envoyproxy/go-control-plane/envoy/service/ext_proc/v3"
+
+	"github.com/envoyproxy/ai-gateway/internal/apischema/openai"
+)
+
+const (
+	sseEventPrefix = "event:"
+	sseDataPrefix  = "data: "
+	sseDoneMessage = "[DONE]"
+)
+
+// streamingToolCall holds the state for a single tool call that is being streamed.
+type streamingToolCall struct {
+	ID         string
+	Name       string
+	InputJSON  bytes.Buffer
```

Comment:
i don't see any reason to use `bytes.Buffer` which results in unnecessarily copying string twice from `anthropic.ContentBlockDeltaEvent.PartialJson` to `bytes.Buffer ([]byte)` and eventually into a string again via `bytes.Buffer.String()`.
```suggestion
	InputJSON  string
```